src/core/dev_api/openvino/core/so_extension.hpp

```diff
@@ -40,6 +41,11 @@ inline std::string resolve_extension_path(const std::string& path) {
 }
 
 inline std::vector<Extension::Ptr> load_extensions(const std::string& path) {
+#ifndef __EMSCRIPTEN__
+    if (ov::util::is_symlink_or_hardlink(path)) {
+        OPENVINO_THROW("Cannot add extension " + path + ". The extension library path is a symlink or hardlink");
+    }
```

Comment:
Prefer `OPENVINO_ASSERT` in such cases:
```suggestion
    OPENVINO_ASSERT(!ov::util::is_symlink_or_hardlink(path),
        "Cannot add extension ", path, ". The extension library path is a symlink or hardlink");
    }
```
Comment applies to similar code down below.